lib/utils.js

```diff
@@ -2,22 +2,48 @@
 
 const { request } = require('undici')
 
-async function refreshAccessToken ({ refreshEndpoint, refreshToken, clientId }) {
-  const { statusCode, body } = await request(`${refreshEndpoint}/token`, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json'
-    },
-    body: JSON.stringify({
+async function refreshAccessToken ({ idpTokenUrl, refreshToken, clientId, clientSecret, contentType }) {
+  let objToSend = null
+
+  if (refreshToken) {
+    objToSend = {
       refresh_token: refreshToken,
       grant_type: 'refresh_token',
       client_id: clientId
-    })
+    }
+  } else {
+    objToSend = {
+      grant_type: 'client_credentials',
+      client_id: clientId,
+      client_secret: clientSecret
+    }
+  }
+
+  let bodyToSend
+
+  if (contentType === 'json') {
+    contentType = 'application/json'
+    bodyToSend = JSON.stringify(objToSend)
+  } else {
+    contentType = 'application/x-www-form-urlencoded'
+    const params = new URLSearchParams()
+    for (const [key, value] of Object.entries(objToSend)) {
+      params.set(key, value)
+    }
+    bodyToSend = params.toString()
+  }
+
+  const { statusCode, body } = await request(idpTokenUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': contentType
```

Comment:
For interoperability purpose also include `accept: 'application/json'` otherwise some providers, e.g. github, will give you XML, known bug, reported a number of times.